### DISCUSSION TITLE
Constant vertex colour display

### DISCUSSION THREAD

**Discussion by fabal** [Autodesk/maya-usd#1501] [Q&A]
Hi there, the VP2 render delegate displays vertex colours authored in USD using a constant interpolation differently than other interpolation, i.e.:
https://github.com/Autodesk/maya-usd/blob/7250d4adbd08f212188e5e211215d02f4498e1ae/lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp#L1784
The rationale for this behaviour is unclear to me, is it still needed?

**Comment by kxl-adsk** [comment]
Hi @fabal, I will convert this issue to a discussion and in future please use discussion forums to ask question.

**Comment by williamkrick** [comment]
Hi @fabal,

Yes, the behavior of the code here is desired and necessary.

Vp2RenderDelegate uses a different shader to draw the object when the color is constant across the whole object ([HdInterpolationConstant & HdInterpolationInstance](https://www.sidefx.com/docs/hdk/hd_2enums_8h.html#a99f077deb5e7ab49413d124d01f66974)) vs. when the color varies across the object. The shader returned by GetFallbackShader has a non-varying float4 color parameter, while the shader returned by GetFallbackCPVShader has a varying float4 color parameter, so the Color Per Vertex material has an additional color geometry stream requirement.



If we used the CPV shader for both type of interpolation then when the interpolation is constant we'd have to create a color buffer and fill every value with the same color, which is wasteful.

**Reply by fabal** [reply]
Thanks for the details. To give you more context, the issue is that the displayed colour, if it is constant, is different than if it is varying.
I can see that in the constant case, a colour correction is applied, could it be that the same doesn't happen in the case of the CPV shader?

**Comment by fabal** [comment]
Just found out the `PIXMAYA_LINEAR_COLORS` environment variable which does what's needed.
